[python3] build failure with linux dynamic triplet

**Host Environment**
- OS: Ubuntu 19.10
- Compiler: GCC v9.2.1

**To Reproduce**
copy/configure a dynamic triplet for linux.
[x64-linux-shared.txt](https://github.com/microsoft/vcpkg/files/4559335/x64-linux-shared.txt)
Steps to reproduce the behavior:
`./vcpkg install python3[enable-shared]:x64-linux-shared`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: make install -j 9
    Working Directory: /home/aaron/dev/vcpkg-python/vcpkg/buildtrees/python3/src/v3.7.3-1161ad7a91-x64-linux-shared-release
    See logs for more information:
      /home/aaron/dev/vcpkg-python/vcpkg/buildtrees/python3/make-build-x64-linux-shared-release-out.log
      /home/aaron/dev/vcpkg-python/vcpkg/buildtrees/python3/make-build-x64-linux-shared-release-err.log

Call Stack (most recent call first):
  ports/python3/portfile.cmake:141 (vcpkg_execute_build_process)
  scripts/ports.cmake:90 (include)


Error: Building package python3:x64-linux-shared failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: python3:x64-linux-shared
  Vcpkg version: 2020.02.04-unknownhash
```
[make-build-x64-linux-shared-release-err.log](https://github.com/microsoft/vcpkg/files/4559343/make-build-x64-linux-shared-release-err.log)


**Additional context**
It appears zlib is missing when attempting to build python3 as shared. I resolved this locally by adding zlib to ports/python3/CONTROL as a dependency.

--- a/ports/python3/CONTROL
+++ b/ports/python3/CONTROL
@@ -1,8 +1,9 @@
 Source: python3
-Version: 3.7.3-2
+Version: 3.7.3-3
 Homepage: https://github.com/python/cpython
 Description: The Python programming language as an embeddable library
 Build-Depends: libffi, openssl
 
 Feature: enable-shared
 Description: Build shared libraries in addition to static ones built by default
+Build-Depends: zlib (!uwp&!windows)

--- a/ports/python3/portfile.cmake
+++ b/ports/python3/portfile.cmake
@@ -18,6 +18,9 @@ vcpkg_from_github(
 
 if("enable-shared" IN_LIST FEATURES)
 	set(_ENABLED_SHARED --enable-shared)
+    if(VCPKG_TARGET_IS_LINUX)
+        message(WARNING"Feature enable-shared requires libffi-devel from the system package manager, please install it on Ubuntu system via sudo apt-get install libffi-dev.")
+    endif()
 else()
 	unset(_ENABLED_SHARED)
 endif()

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -1092,6 +1092,10 @@ miniupnpc:arm-uwp=fail
 miniupnpc:x64-uwp=fail
 minizip:arm-uwp=fail
 minizip:x64-uwp=fail
+# conflicts with stb
+mlpack:x86-windows=skip
+mlpack:x64-windows=skip
+mlpack:x64-windows-static=skip
 mman:x64-linux=fail
 mman:x64-osx=fail
 # mmx installs many problematic headers, such as `json.h` and `sched.h`

